homeassistant/components/sensor/recorder.py

```diff
@@ -189,18 +181,31 @@ def _time_weighted_average(
     return accumulated / (end - start).total_seconds()
 
 
+def _units_same(fstates: list[tuple[float, State]]) -> bool:
+    """Return True if all states have the same unit."""
+    states = next(itertools.islice(zip(*fstates), 1, 2))
+    units = iter(state.attributes.get(ATTR_UNIT_OF_MEASUREMENT) for state in states)
+    first = next(units)
+    return all(first == x for x in units)
```

Comment:
A slightly more readable alternative:
```suggestion
    return len({item[1].attributes.get(ATTR_UNIT_OF_MEASUREMENT) for item in fstates}) < 2
```

Reply:
Fixed